sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricsEnvironment.java

```diff
@@ -52,8 +53,8 @@
   private static final AtomicBoolean METRICS_SUPPORTED = new AtomicBoolean(false);
   private static final AtomicBoolean REPORTED_MISSING_CONTAINER = new AtomicBoolean(false);
 
-  private static final ThreadLocal<@Nullable MetricsContainer> CONTAINER_FOR_THREAD =
-      new ThreadLocal<>();
+  private static final ThreadLocal<@Nullable MetricsContainerHolder> CONTAINER_FOR_THREAD =
```

Comment:
Is it always no longer `@Nullable`? Could possible localize `@SuppressWarnings` to this initialization line? (just an idea)

Reply:
Yes due to the withInitialValue, however I believe I tried it and it didn't seem to work without the Nullable.  I think it didn't like it later with the get method